package main

import (
	"errors"
	"fmt"
)

func somaDigitos(numero int) (int, error) {
	if numero < 0 {
		return 0, errors.New("Número negativo. Não é possível calcular a soma dos dígitos.")
	}

	soma := 0
	for numero != 0 {
		digito := numero % 10
		soma += digito
		numero /= 10
	}

	return soma, nil
}

func main() {
	numero := 12345

	resultado, err := somaDigitos(numero)

	if err != nil {
		fmt.Println("Erro:", err)
	} else {
		fmt.Printf("A soma dos dígitos de %d é: %d\n", numero, resultado)
	}
}

